packages/core/logging/core-logging-server-internal/src/appenders/rolling_file/rolling_file_context.ts

```diff
@@ -36,4 +38,17 @@ export class RollingFileContext {
       this.currentFileSize = 0;
     }
   }
+
+  #orderedRolledFileFn?: GetOrderedRolledFileFn;
+
+  public async getOrderedRolledFiles(): Promise<string[]> {
+    if (this.#orderedRolledFileFn) {
+      return this.#orderedRolledFileFn();
+    }
+    throw new Error('orderedRolledFileFn not registered on the context');
+  }
+
+  public setOrderedRolledFileFn(fn: GetOrderedRolledFileFn) {
+    this.#orderedRolledFileFn = fn;
+  }
```

Comment:
Using the rolling context as communication mechanism between the rolling strategy and the retention policy. FWIW, this is very close to how log4j does it.